test/server/cards/03_TWI/leaders/NuteGunrayVindictiveViceroy.spec.ts

```diff
@@ -0,0 +1,58 @@
+describe('Nute Gunray, Vindictive Viceroy', function () {
+    integration(function (contextRef) {
+        describe('Nute Gunray\'s leader undeployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine'],
+                        spaceArena: ['green-squadron-awing'],
+                        leader: 'nute-gunray#vindictive-viceroy',
+                        resources: 4,
+                    },
+                    player2: {
+                        hand: ['admiral-yularen#advising-caution', 'superlaser-blast'],
+                    },
+                });
+            });
+
+            it('should create a battledroid token if two or more friendly units defeated this phase', function () {
+                const { context } = contextRef;
+
+                context.player1.passAction();
```

Comment:
If you need player2 to start you can set `hasInitiative: true` as well (in the same place you set `resources: 4` above)